girder/utility/s3_assetstore_adapter.py

```diff
@@ -191,6 +191,12 @@ def initUpload(self, upload):
         if upload['size'] <= 0:
             return upload
 
+        # collapse consecutive spaces in the filename into a single space
+        # this is due to a bug in S3 that does not properly handle filenames
+        # with multiple spaces in a row, resulting in a SignatureDoesNotMatch
+        # error
+        upload['name'] = ' '.join(filter(None, upload['name'].split(' ')))
```

Comment:
I just tested, and S3 actually supports names with both leading and trailing spaces, so I would propose a slightly less restrictive name modification:

    upload['name'] = re.sub('\s+', ' ', upload['name'])

In case some crazy person really wants files with spaces at the beginning or end.